app-modules/campaign/src/Filament/Resources/CampaignResource/RelationManagers/CampaignActionsRelationManager.php

```diff
@@ -77,11 +79,20 @@ public function form(Form $form): Form
 
     public function table(Table $table): Table
     {
+        /** @var Campaign $campaign */
+        $campaign = $this->getOwnerRecord();
+
         return $table
             ->recordTitleAttribute('id')
             ->modifyQueryUsing(fn (QueryBuilder $query) => $query->orderBy('execute_at', 'ASC'))
             ->columns([
                 TextColumn::make('type')->label('Step Type'),
+                TextColumn::make('cancelled_at')
+                    ->label('')
+                    ->formatStateUsing(fn () => 'Cancelled')
+                    ->color('danger')
+                    ->hidden(fn (?CampaignAction $record) => ! CancelCampaignAction::active() || $record?->cancelled_at !== null)
```

Comment:
I could be wrong here but, can `$record` even be `nullable`? It wouldn't make sense for a row to exist and call this function if it wasn't defined.

```suggestion
                    ->hidden(fn (CampaignAction $record) => ! CancelCampaignAction::active() || $record->cancelled_at !== null)
```

Reply:
Hmm, got it. I guess Filament may be loading in a null row. Alright.